FunctionalTableData/TableItemLayout.swift

```diff
@@ -14,44 +14,117 @@ public protocol TableItemLayout {
 
 // TableItemLayout
 
-public typealias EdgeBasedTableItemLayout = CombinedLayout<EdgeLayout.Horizontal, EdgeLayout.Vertical>
-public typealias LayoutMarginsTableItemLayout = CombinedLayout<MarginsLayout.Horizontal, MarginsLayout.Vertical>
+public typealias EdgeBasedTableItemLayout = Layout<EdgeLayout.Top, EdgeLayout.Leading, EdgeLayout.Bottom, EdgeLayout.Trailing>
+public typealias LayoutMarginsTableItemLayout = Layout<MarginsLayout.Top, MarginsLayout.Leading, MarginsLayout.Bottom, MarginsLayout.Trailing>
 
 public struct ExplicitLayoutMarginsTableItemLayout: TableItemLayout {
 	public static func layoutView(_ view: UIView, inContentView contentView: UIView) {
 		contentView.preservesSuperviewLayoutMargins = false
-		view.constrainToFillView(contentView, respectingLayoutMargins: true)
+		
+		MarginsLayout.Top.layoutView(view, inContentView: contentView)
+		MarginsLayout.Leading.layoutView(view, inContentView: contentView)
+		MarginsLayout.Bottom.layoutView(view, inContentView: contentView)
+		MarginsLayout.Trailing.layoutView(view, inContentView: contentView)
 	}
 }
 
 public enum EdgeLayout {
+	@available(*, deprecated, message: "Use `EdgeLayout.Leading` and `EdgeLayout.Trailing` instead.")
 	public struct Horizontal: TableItemLayout {
 		public static func layoutView(_ view: UIView, inContentView contentView: UIView) {
-			view.constrainToFillViewHorizontally(contentView)
+			Leading.layoutView(view, inContentView: contentView)
+			Trailing.layoutView(view, inContentView: contentView)
 		}
 	}
+	
+	@available(*, deprecated, message: "Use `EdgeLayout.Top` and `EdgeLayout.Bottom` instead.")
 	public struct Vertical: TableItemLayout {
 		public static func layoutView(_ view: UIView, inContentView contentView: UIView) {
-			view.constrainToFillViewVertically(contentView)
+			Top.layoutView(view, inContentView: contentView)
+			Bottom.layoutView(view, inContentView: contentView)
+		}
+	}
+	
+	public struct Top: TableItemLayout {
+		public static func layoutView(_ view: UIView, inContentView contentView: UIView) {
+			view.topAnchor.constraint(equalTo: contentView.topAnchor).isActive = true
+		}
+	}
+	
+	public struct Leading: TableItemLayout {
+		public static func layoutView(_ view: UIView, inContentView contentView: UIView) {
+			view.leadingAnchor.constraint(equalTo: contentView.leadingAnchor).isActive = true
+		}
+	}
+	
+	public struct Bottom: TableItemLayout {
+		public static func layoutView(_ view: UIView, inContentView contentView: UIView) {
+			contentView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
```

Comment:
It's awkward that these definitions flip the order of `view` and `contentView` for the bottom and trailing constraints. All four should be consistently defined.

Reply:
> It's awkward that these definitions flip the order of view and contentView for the bottom and trailing constraints. All four should be consistently defined.

This is how Interface Builder does it as well (flipping the order), otherwise we will have to `* -1` any value passed to it.

In this cases is safe because there is no constant added

Reply:
Had not noticed that IB was doing that too 🤔 I guess for the same reason you describe.